addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/httputils/HtmlContext.java

```diff
@@ -40,9 +41,11 @@ public class HtmlContext {
     private int end = 0;
     private List<String> parentTags = new ArrayList<>();
     private String tagAttribute = null;
+    private HashMap<String, Boolean> tagAttributes = new HashMap<>();
```

Comment:
Why a map?

Reply:
The payload I have used uses two attributes combined... So inorder to catch both the attributes and the names I have to use a map :). In the website the alert dialog box comes up when the user hits alt+x or something like that :)

Reply:
I understand why the need but not why a map, wouldn't a list be enough? It's just querying for the presence of the attributes not their values or other state (it's always setting the value to true when present).

Reply:
You're right .... Actually a better one would be a proper key:value map instead of the key:boolean map though... So what should I do? List or a map?

Reply:
A map with name/value sounds good.